### PR TITLE
[DEPENDENCIES] Upgrade Kotlin version to 2.0.21 for #1897, cleanup dependencies, upgrade Milvus and Guava

### PR DESCRIPTION
**Pull request by kpavlov**
## Issue #1897 

Dependency upgrade before  #1897 

## Change

- Use Kotlin BOM for dependency resolution
- Remove explicit Kotlin dependencies from multiple modules
   This eliminates redundant Kotlin standard library dependencies across various `pom.xml` files. The change resolves potential conflicts and reduces unnecessary dependency overhead.
* Updated `guava` version from `32.0.0-jre` to `32.1.2-jre` in `langchain4j-zhipu-ai/pom.xml`.
* Updated Milvus SDK to version 2.4.2 

## General checklist

- [x] There are no breaking changes
- [ ] I have added unit and integration tests for my change
- [ ] I have manually run all the unit and integration tests in the module I have added/changed, and they are all green
- [ ] I have manually run all the unit and integration tests in the [core](https://github.com/langchain4j/langchain4j/tree/main/langchain4j-core) and [main](https://github.com/langchain4j/langchain4j/tree/main/langchain4j) modules, and they are all green

- [ ] I have added/updated the [documentation](https://github.com/langchain4j/langchain4j/tree/main/docs/docs)
- [ ] I have added an example in the [examples repo](https://github.com/langchain4j/langchain4j-examples) (only for "big" features)
- [ ] I have added/updated [Spring Boot starter(s)](https://github.com/langchain4j/langchain4j-spring) (if applicable)

## Checklist for adding new maven module

- [ ] I have added my new module in the root `pom.xml` and `langchain4j-bom/pom.xml`

## Checklist for adding new embedding store integration

- [ ] I have added a `{NameOfIntegration}EmbeddingStoreIT` that extends from either `EmbeddingStoreIT` or `EmbeddingStoreWithFilteringIT`
- [ ] I have added a `{NameOfIntegration}EmbeddingStoreRemovalIT` that extends from `EmbeddingStoreWithRemovalIT`

## Checklist for changing existing embedding store integration

- [ ] I have manually verified that the `{NameOfIntegration}EmbeddingStore` works correctly with the data persisted using the latest released version of LangChain4j

### CODE REVIEW

**Comment by langchain4j** [comment]
@kpavlov thank you! Some modules still use `kotlin-stdlib-jdk8` (in this PR), is it on putpose?

A few things to make sure we are on the same page:
- `kotlin-stdlib-jdk8` is currently used only for okhttp and we plan to replace okhttp with jdk 11 httpclient by default, although okhttp could be used as well
- Does okhttp/retrofit works fine with `kotlin-stdlib` instead of `kotlin-stdlib-jdk8`?
- I cannot commit to keep Kotlin-related stuff in this repository, I need to understand the scope first

**Comment by kpavlov** [comment]
Kotlin std-library dependency is `kotlin-stdlib`. `kotlin-stdlib-jdk8` should be used only on jdk 8 with Kotlin < 1.8. 
Since kotlin-bom is upgraded, it will get `kotlin-stdlib-jdk8` 2.0.21 automatically.
https://kotlinlang.org/docs/maven.html#set-dependencies

> Some modules still use kotlin-stdlib-jdk8 (in this PR), is it on putpose?

Nope, I didn't notice it, will fix.

I don't understand, why this dependency was added? In compile scope. There are no references to Kotlin stdlib code

> Does okhttp/retrofit works fine with kotlin-stdlib instead of kotlin-stdlib-jdk8?

That's the unit and integration testing for. `kotlin-stdlib-jdk8` will be used as transitive dependency.

**Comment by kpavlov** [comment]
Let's merge #1913 first, I am not feeling comfortable to continue without tests

File: langchain4j-jlama/pom.xml [hunk 1]
```diff
@@ -81,17 +82,6 @@
             <type>test-jar</type>
             <scope>test</scope>
         </dependency>
-
-        <dependency>
-            <groupId>org.mockito</groupId>
-            <artifactId>mockito-core</artifactId>
```
Comment (langchain4j) [inline langchain4j-jlama/pom.xml:91]: Why removing these? They are required for `JlamaAiServicesWithToolsIT`

**Comment by langchain4j** [comment]
> I don't understand, why this dependency was added? In compile scope. There are no references to Kotlin stdlib code

https://github.com/langchain4j/langchain4j/issues/1506

**Comment by langchain4j** [comment]
> That's the unit and integration testing for.

Tests can miss something, it is never a bad idea to double-check

**Review by ghost — COMMENTED** [review]
@katcharov thanks!

File: langchain4j-jlama/pom.xml (hunk 1 above)
Comment (kpavlov) [reply langchain4j-jlama/pom.xml:91]: As Java 21 and 23 builds are green, it means it's safe to removed this dependency here.
It is now transitive from [parent pom](https://github.com/langchain4j/langchain4j/blob/main/langchain4j-parent/pom.xml#L363-L369).

**Comment by langchain4j** [comment]
@kpavlov there are conflicts now, sorry about that

**Review by ghost — APPROVED** [review]
@kpavlov thank you!

**Pull request merged**